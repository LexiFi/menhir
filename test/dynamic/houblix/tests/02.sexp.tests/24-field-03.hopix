let t = { a = "0", x = "Field extraction is still a success!\n", y = 31 }
let u = { a = { a = { a = t } }, b = 42 }
let z = u.a.a.a.x
let test = print_string (z)