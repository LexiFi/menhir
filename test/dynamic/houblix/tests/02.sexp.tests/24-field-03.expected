(((value(DefineValue(SimpleValue(((value(Id t))(position((start_p(""1 0 4))(end_p(""1 0 5)))))()((value(Record((((value(LId a))(position((start_p(""1 0 10))(end_p(""1 0 11)))))((value(Literal((value(LString 0))(position((start_p(""1 0 16))(end_p(""1 0 17)))))))(position((start_p(""1 0 16))(end_p(""1 0 17))))))(((value(LId x))(position((start_p(""1 0 19))(end_p(""1 0 20)))))((value(Literal((value(LString"Field extraction is still a success!\n"))(position((start_p(""1 0 62))(end_p(""1 0 63)))))))(position((start_p(""1 0 62))(end_p(""1 0 63))))))(((value(LId y))(position((start_p(""1 0 65))(end_p(""1 0 66)))))((value(Literal((value(LInt 31))(position((start_p(""1 0 69))(end_p(""1 0 71)))))))(position((start_p(""1 0 69))(end_p(""1 0 71)))))))()))(position((start_p(""1 0 8))(end_p(""1 0 73)))))))))(position((start_p(""1 0 0))(end_p(""1 0 73)))))((value(DefineValue(SimpleValue(((value(Id u))(position((start_p(""2 74 78))(end_p(""2 74 79)))))()((value(Record((((value(LId a))(position((start_p(""2 74 84))(end_p(""2 74 85)))))((value(Record((((value(LId a))(position((start_p(""2 74 90))(end_p(""2 74 91)))))((value(Record((((value(LId a))(position((start_p(""2 74 96))(end_p(""2 74 97)))))((value(Variable((value(Id t))(position((start_p(""2 74 100))(end_p(""2 74 101)))))()))(position((start_p(""2 74 100))(end_p(""2 74 101)))))))()))(position((start_p(""2 74 94))(end_p(""2 74 103)))))))()))(position((start_p(""2 74 88))(end_p(""2 74 105))))))(((value(LId b))(position((start_p(""2 74 107))(end_p(""2 74 108)))))((value(Literal((value(LInt 42))(position((start_p(""2 74 111))(end_p(""2 74 113)))))))(position((start_p(""2 74 111))(end_p(""2 74 113)))))))()))(position((start_p(""2 74 82))(end_p(""2 74 115)))))))))(position((start_p(""2 74 74))(end_p(""2 74 115)))))((value(DefineValue(SimpleValue(((value(Id z))(position((start_p(""3 116 120))(end_p(""3 116 121)))))()((value(Field((value(Field((value(Field((value(Field((value(Variable((value(Id u))(position((start_p(""3 116 124))(end_p(""3 116 125)))))()))(position((start_p(""3 116 124))(end_p(""3 116 125)))))((value(LId a))(position((start_p(""3 116 126))(end_p(""3 116 127)))))))(position((start_p(""3 116 124))(end_p(""3 116 127)))))((value(LId a))(position((start_p(""3 116 128))(end_p(""3 116 129)))))))(position((start_p(""3 116 124))(end_p(""3 116 129)))))((value(LId a))(position((start_p(""3 116 130))(end_p(""3 116 131)))))))(position((start_p(""3 116 124))(end_p(""3 116 131)))))((value(LId x))(position((start_p(""3 116 132))(end_p(""3 116 133)))))))(position((start_p(""3 116 124))(end_p(""3 116 133)))))))))(position((start_p(""3 116 116))(end_p(""3 116 133)))))((value(DefineValue(SimpleValue(((value(Id test))(position((start_p(""4 134 138))(end_p(""4 134 142)))))()((value(Apply((value(Variable((value(Id print_string))(position((start_p(""4 134 145))(end_p(""4 134 157)))))()))(position((start_p(""4 134 145))(end_p(""4 134 157)))))((value(Variable((value(Id z))(position((start_p(""4 134 159))(end_p(""4 134 160)))))()))(position((start_p(""4 134 158))(end_p(""4 134 161)))))))(position((start_p(""4 134 145))(end_p(""4 134 161)))))))))(position((start_p(""4 134 134))(end_p(""4 134 161))))))
